proxy/src/fs_watch.rs

```diff
@@ -121,15 +114,49 @@ impl PathAndHash {
         }
     }
 
-    fn has_changed(&self) -> io::Result<bool> {
-        let contents = fs::read(&self.path)?;
-        let hash = Some(digest::digest(&digest::SHA256, &contents[..]));
-        let changed = self.last_hash.borrow().as_ref().map(Digest::as_ref)
-            != hash.as_ref().map(Digest::as_ref);
-        if changed {
-            self.last_hash.replace(hash);
+    /// Attempts to update the hash for this file and checks if it has changed.
+    ///
+    /// # Returns
+    /// - `Ok(true)` if the file was read successfully and the hash has changed.
+    /// - `Ok(false)` if we were able to read the file but the has has not
+    ///    changed.
+    /// - `Err(io::Error)` if an error occurred reading the file.
+    fn update_and_check(&self) -> io::Result<bool> {
+        match fs::read(&self.path) {
+            Ok(contents) => {
+                // If we were able to read the file, compare the hash of its
+                // current contents with the previous hash to determine if it
+                // has changed.
+                let hash = Some(digest::digest(&digest::SHA256, &contents[..]));
+                let changed = self.last_hash
+                    .borrow()
+                    .as_ref()
+                    .map(Digest::as_ref)
+                    // Compare the last hash with the current hash.
+                    .ne(&hash.as_ref().map(Digest::as_ref));
```

Comment:
Can this be written as:

The "compare" comment seems like a bit of a red flag to me -- it's an extremely obvious comment, but it's necessary here because it's a bit hard to read otherwise. This makes me think we should make it more readable so the comment can go away.  Something more like:
```rust
let prior_hash = Digest::as_ref(&digest::digest(&digest::SHA256, &contents[..]);
let current_hash = self.last_hash.borrow().as_ref().map(Digest::as_ref);

let changed = prior_hash.ne(current_hash.as_ref());
if changed {
}
Ok(changed))
```

Reply:
Unfortunately, if I recall correctly, the borrow checker wasn't a fan of doing it like that.